File: Swift-C++-IR-Comparison/test/functions/Swift/FunctionTypesAsParameter.swift
@main
struct MyMain {

    static func main() {
        var mathFunction: (Int, Int) -> Int = addTwoInts
        var result = takeMathFunction(mathFunction, 42, 44)
    }

}

func takeMathFunction(_ mathFunction: (Int, Int) -> Int, _ a: Int, _ b: Int) {
    let exec = mathFunction(a, b)
}

func addTwoInts(_ a: Int, _ b: Int) -> Int {
    return a + b
}

File: Swift-C++-Testsuite/test/functions/C++/FunctionTypesAsParameter.cpp

int takeMathFunction(int(*mathFunction)(int, int), int a, int b) {
   int exec = mathFunction(a, b);
   return exec;
}

int addTwoInts(int a, int b) {
	return a + b;
}

int main() {
	int(*mathFunction)(int, int);
	mathFunction = addTwoInts;
	int result = takeMathFunction(mathFunction, 42, 44);
	return 0;
	
}